### ISSUE TITLE
NO change

### ISSUE THREAD

**Issue by ZhenboZhao77** [nizhenliang/CGRSeg#8]
Dear Author,

I am very excited to see your masterpiece and couldn't wait to add the RCM module to my model. The code is as follows:
from .rcm import RCM, RCA
self.rcm = RCM(dim=256)
x = self.rcm(x)
However, I found that the model didn't seem to change. It neither improved nor worsened. Did I use the RCM correctly, or do I need to adjust some hyperparameters?

**Comment by ZhenboZhao77** [comment]
hello,can you help me?

**Comment by nizhenliang** [comment]
You can adjust the training settings, such as the learning rate and training policy. The location added to the network will also have an effect. You can try to adjust it. The number of channels can also be adjusted to fit your model.

**Comment by nizhenliang** [comment]
> Dear Author,
> 
> I am very excited to see your masterpiece and couldn't wait to add the RCM module to my model. The code is as follows: from .rcm import RCM, RCA self.rcm = RCM(dim=256) x = self.rcm(x) However, I found that the model didn't seem to change. It neither improved nor worsened. Did I use the RCM correctly, or do I need to adjust some hyperparameters?

You can adjust the training settings, such as the learning rate and training policy. The location added to the network will also have an effect. You can try to adjust it. The number of channels can also be adjusted to fit your model.

**Comment by Billy-ZTB** [comment]
Do you see any improvement now? Where did you put this module in?

**Comment by Billy-ZTB** [comment]
> > Dear Author,
> > I am very excited to see your masterpiece and couldn't wait to add the RCM module to my model. The code is as follows: from .rcm import RCM, RCA self.rcm = RCM(dim=256) x = self.rcm(x) However, I found that the model didn't seem to change. It neither improved nor worsened. Did I use the RCM correctly, or do I need to adjust some hyperparameters?
> 
> You can adjust the training settings, such as the learning rate and training policy. The location added to the network will also have an effect. You can try to adjust it. The number of channels can also be adjusted to fit your model.

Hello! Should the RCM act like the way in your article which uses multiple RCMs and multiplied in a recurrent way? I only used one RCM after the encoder to aggregate features in the last 3 stages of encoder, and my model is like the Unet, but it didn't show improvements. But it confuses me when I try to use the similar architecture in unet.

**Comment by nizhenliang** [comment]
Yes, you should use multiple RCMs.  One RCM is not enough to extract multi-scale contexts well.

**Comment by Billy-ZTB** [comment]
Thanks! I will try it later.



---- Replied Message ----
| From | ***@***.***> |
| Date | 10/15/2024 10:26 |
| To | ***@***.***> |
| Cc | Zhang ***@***.***>***@***.***> |
| Subject | Re: [nizhenliang/CGRSeg] NO change (Issue #8) |

Yes, you should use multiple RCMs. One RCM is not enough to extract multi-scale contexts well.
